src/access.rs

```diff
@@ -0,0 +1,378 @@
+				c.transactions_confirmed(&block_header, &[(pos, &tx)], block_height);
+			}
+		}
+
+		*self.queued_transactions.lock().unwrap() = Vec::new();
+		*self.watched_transactions.lock().unwrap() = unconfirmed_registered_txs;
+		*self.queued_outputs.lock().unwrap() = Vec::new();
+		*self.watched_outputs.lock().unwrap() = unspent_registered_outputs;
+
+		Ok(())
+	}
+
+	async fn sync_transaction_unconfirmed(
+		&self, confirmables: &Vec<&(dyn Confirm + Sync)>,
+	) -> Result<(), Error> {
+		let client = &*self.blockchain;
+		// Query the interface for relevant txids and check whether they have been
+		// reorged-out of the chain.
+		let relevant_txids =
+			confirmables.iter().flat_map(|c| c.get_relevant_txids()).collect::<HashSet<Txid>>();
+		for txid in relevant_txids {
+			let tx_unconfirmed = client
+				.get_tx_status(&txid)
+				.await
+				.ok()
+				.unwrap_or(None)
+				.map_or(true, |status| !status.confirmed);
+			if tx_unconfirmed {
+				for c in confirmables {
+					c.transaction_unconfirmed(&txid);
+				}
+			}
+		}
+
+		Ok(())
+	}
+
+	pub(crate) async fn create_funding_transaction(
+		&self, output_script: &Script, value_sats: u64, confirmation_target: ConfirmationTarget,
+	) -> Result<Transaction, Error> {
+		let num_blocks = num_blocks_from_conf_target(confirmation_target);
+		let fee_rate = self.blockchain.estimate_fee(num_blocks).await?;
+
+		let locked_wallet = self.wallet.lock().unwrap();
+		let mut tx_builder = locked_wallet.build_tx();
+
+		tx_builder.add_recipient(output_script.clone(), value_sats).fee_rate(fee_rate).enable_rbf();
+
+		let (mut psbt, _) = tx_builder.finish()?;
+		log_trace!(self.logger, "Created funding PSBT: {:?}", psbt);
+
+		// We double-check that no inputs try to spend non-witness outputs. As we use a SegWit
+		// wallet descriptor this technically shouldn't ever happen, but better safe than sorry.
+		for input in &psbt.inputs {
+			if input.witness_utxo.is_none() {
+				log_error!(self.logger, "Tried to spend a non-witness funding output. This must not ever happen. Panicking!");
+				panic!("Tried to spend a non-witness funding output. This must not ever happen.");
+			}
+		}
+
+		if !locked_wallet.sign(&mut psbt, SignOptions::default())? {
+			return Err(Error::FundingTxCreationFailed);
+		}
+
+		Ok(psbt.extract_tx())
+	}
+
+	pub(crate) fn get_new_address(&self) -> Result<bitcoin::Address, Error> {
+		let address_info = self.wallet.lock().unwrap().get_address(AddressIndex::New)?;
+		Ok(address_info.address)
+	}
+}
+
+impl<D> FeeEstimator for ChainAccess<D>
+where
+	D: BatchDatabase,
+{
+	fn get_est_sat_per_1000_weight(&self, confirmation_target: ConfirmationTarget) -> u32 {
+		let num_blocks = num_blocks_from_conf_target(confirmation_target);
+		let fallback_fee = fallback_fee_from_conf_target(confirmation_target);
+
+		let locked_runtime = self.tokio_runtime.read().unwrap();
+		if locked_runtime.as_ref().is_none() {
+			return fallback_fee;
+		}
+
+		locked_runtime.as_ref().unwrap().block_on(async {
```

Comment:
Got rid of most of them, one non-trivial left for the moment.

Reply:
Might be blocked on https://github.com/bitcoindevkit/bdk/issues/795, otherwise we need to run all wallet operations in it's own thread and keep it `block_on`.